src/main/kotlin/io/kotlintest/TestRunner.kt

```diff
@@ -0,0 +1,62 @@
+package io.kotlintest
+
+import org.junit.platform.engine.EngineExecutionListener
+import org.junit.platform.engine.TestExecutionResult
+import org.junit.runners.model.TestTimedOutException
+import java.util.concurrent.Executors
+
+class TestRunner(val listener: EngineExecutionListener) {
+
+  private fun <CONTEXT> createInterceptorChain(
```

Comment:
Maybe we can make this a top level function and reuse it for test cases, specs and the whole project.